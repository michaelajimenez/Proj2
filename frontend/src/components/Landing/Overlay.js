import React from "react";
export default function Overlay({ togglePanel }) {
  return (
    <div className="overlay-container">
      <div className="overlay">
        <div className="overlay-panel overlay-left">
          <h1>Have an Account?</h1>
          <p className="overlay-description">
            How about signing back in to your current?
          </p>
          <button
            className="overlayBtn"
            onClick={() => {
              togglePanel(false);
            }}>
            Sign In
          </button>
        </div>
        <div className="overlay-panel overlay-right">
          <h1>Need an Account?</h1>
          <p className="overlay-description">Click below to sign up today!</p>

          <button
            className="overlayBtn"
            onClick={() => {
              togglePanel(true);
            }}>
            Sign Up
          </button>
        </div>
      </div>
    </div>
  );
}
