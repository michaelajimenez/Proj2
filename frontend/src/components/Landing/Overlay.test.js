import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from "react-dom";
import Overlay from './Overlay.js';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


test('Overlay Component renders properly',()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Overlay/>,div)
});

test('component rerenders when props change',()=>{
    const component = renderer.create(<Overlay/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.togglePanel = true;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
    tree.props.togglePanel = false;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Overlay Component contains One div named overlay-container', ()=>{
    let component = shallow(<Overlay/>);
    expect(component.find("div.overlay-container")).toHaveLength(1);
});