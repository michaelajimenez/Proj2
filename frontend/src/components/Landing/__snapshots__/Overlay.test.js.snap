// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component rerenders when props change 1`] = `
<div
  className="overlay-container"
>
  <div
    className="overlay"
  >
    <div
      className="overlay-panel overlay-left"
    >
      <h1>
        Have an Account?
      </h1>
      <p
        className="overlay-description"
      >
        How about signing back in to your current?
      </p>
      <button
        className="overlayBtn"
        onClick={[Function]}
      >
        Sign In
      </button>
    </div>
    <div
      className="overlay-panel overlay-right"
    >
      <h1>
        Need an Account?
      </h1>
      <p
        className="overlay-description"
      >
        Click below to sign up today!
      </p>
      <button
        className="overlayBtn"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
  </div>
</div>
`;

exports[`component rerenders when props change 2`] = `
<div
  className="overlay-container"
>
  <div
    className="overlay"
  >
    <div
      className="overlay-panel overlay-left"
    >
      <h1>
        Have an Account?
      </h1>
      <p
        className="overlay-description"
      >
        How about signing back in to your current?
      </p>
      <button
        className="overlayBtn"
        onClick={[Function]}
      >
        Sign In
      </button>
    </div>
    <div
      className="overlay-panel overlay-right"
    >
      <h1>
        Need an Account?
      </h1>
      <p
        className="overlay-description"
      >
        Click below to sign up today!
      </p>
      <button
        className="overlayBtn"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
  </div>
</div>
`;

exports[`component rerenders when props change 3`] = `
<div
  className="overlay-container"
>
  <div
    className="overlay"
  >
    <div
      className="overlay-panel overlay-left"
    >
      <h1>
        Have an Account?
      </h1>
      <p
        className="overlay-description"
      >
        How about signing back in to your current?
      </p>
      <button
        className="overlayBtn"
        onClick={[Function]}
      >
        Sign In
      </button>
    </div>
    <div
      className="overlay-panel overlay-right"
    >
      <h1>
        Need an Account?
      </h1>
      <p
        className="overlay-description"
      >
        Click below to sign up today!
      </p>
      <button
        className="overlayBtn"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
  </div>
</div>
`;
